### PR TITLE
document settingDefaults

### PR DESCRIPTION
**Pull request by bwindels**
See https://github.com/vector-im/riot-web/issues/6791 & https://github.com/vector-im/riot-web/issues/8449

Also https://github.com/vector-im/riot-web/pull/9919

### CODE REVIEW

**Comment by jryans** [comment]
@turt2live How about if we document it here _and_ in riot-web?

**Comment by turt2live** [comment]
I have a fear that they'd get out of sync or forgotten about. Generally I try and consider riot-web to be the config side of things (despite the config being defined here) and the react-sdk being the logic layer. 

I don't really mind if it's both, but at a minimum it should be in riot-web to help users.

**Comment by jryans** [comment]
Hmm, I guess I am confused... The PR here isn't trying to document _each_ setting... It's just trying to mention that settings can be defaulted, which doesn't seem so likely to get out of sync, right? It's not likely to change unless we overhaul settings again.

**Comment by turt2live** [comment]
bleh, yes, that's true. The fact that the object exists though still makes most of my argument make sense, I think. The idea being that "hey, a flag exists for things" should be documented to riot admins and less so developers (ie: dump info in riot-web for riot admins, developer docs in react-sdk).

**Comment by bwindels** [comment]
This one seems to have been forgotten by me. I would go ahead and merge this if nobody objects as this is also already documented in `riot-web/docs/config.md` and in the sample config there. I think the information is also useful for developers though.

File: docs/settings.md [hunk 1]
```diff
@@ -51,6 +51,17 @@ Settings are the different options a user may set or experience in the applicati
 }
 ```
 
+Settings that support the config level can be set in the config file under the `settingDefaults` key (note that the "theme" setting is special cased to the `default_theme` in the config file):
```
Comment (turt2live) [inline docs/settings.md]: There's a couple more like language, etc:
```suggestion
Settings that support the config level can be set in the config file under the `settingDefaults` key (note that some settings, like the "theme" setting, are special cased in the config file):
```

File: docs/settings.md [hunk 2]
```diff
@@ -51,6 +51,17 @@ Settings are the different options a user may set or experience in the applicati
...
+```
+{
+  ...
+  settingDefaults: {
```
Comment (turt2live) [inline docs/settings.md]:
```suggestion
  "settingDefaults": {
```

File: docs/settings.md [hunk 3]
```diff
@@ -51,6 +51,17 @@ Settings are the different options a user may set or experience in the applicati
...
+    settingName: true
```
Comment (turt2live) [inline docs/settings.md]:
```suggestion
    "settingName": true
```

File: docs/settings.md [hunk 4]
```diff
@@ -51,6 +51,17 @@ Settings are the different options a user may set or experience in the applicati
 }
 ```
 
+Settings that support the config level can be set in the config file under the `settingDefaults` key (note that the "theme" setting is special cased to the `default_theme` in the config file):
+```
```
Comment (turt2live) [inline docs/settings.md]:
```suggestion
```json
```

**Comment by t3chguy** [comment]
@bwindels can we get this over the line :D?

**Comment by bwindels** [comment]
Not sure why all these tests are failing, but I am fairly sure it has *nothing* to do with these documentation changes, so merging.

**Pull request merged**